openassessment/xblock/team_mixin.py

```diff
@@ -99,6 +104,41 @@ def valid_access_to_team_assessment(self):
         """
         return self.is_course_staff or self.in_studio_preview or self.has_team()
 
+    def add_team_submission_context(
+        self, context, team_submission_uuid=None, individual_submission_uuid=None, transform_usernames=False
+    ):
+        """
+        Adds team submission information to context dictionary, based on existing team submissions
+        Specifically team name and team_usernames
+
+        Params:
+            - context (dict): a context dict for rendering a page that we will add values to
+            - team_submission_uuid (string): [optional] the uuid of the team submission we want to add context info for
+            - individual_submission_uuid (string): [optional] the uuid of an individual submission that's a part of
+                                                   the team submission for which we want to add context info
+            - transform_usernames (bool): [optional default: False] If False, context['team_usernames'] will be a list
+                                          of username strings. If True, it will be a string, in the form
+                                          "Username1, Username2, ... UsernameN, and UsernameN+1"
+
+        One of team_submission_uuid and individual_submission_uuid are required, and if they are both provided,
+        individual_submission_uuid will be ignored.
+        """
+        if not any((team_submission_uuid, individual_submission_uuid)):
```

Comment:
Would it make sense to have this method return team name for anonymous user (see staff_area_mixin.py::353) ?

Reply:
I'm gonna not do this, in order to keep the two things logically separate. So for now,

get_team_info() gets you the info for the team the user is currently on, 

add_team_submission_context gets you team info specifically from a submission